config/rbac/role.yaml

```diff
@@ -21,12 +21,8 @@ rules:
   resources:
   - operators
   verbs:
-  - create
-  - delete
   - get
   - list
-  - patch
-  - update
```

Comment:
Related but kinda. I still think that "manager-role" is too generic for a `ClusterRole`. Every single operator made with kubebuilder is going to create, by default, a `ClusterRole` named "manager-role", and this _will_ lead to conflicts and debugging issues.

This is apparently done via the Makefile:
https://github.com/kubernetes-sigs/kubebuilder/discussions/2297

Reply:
These changes will not survive a `make manifests`.
AFAICT, the proper mechanism is to update the kubebuilder tags:
https://github.com/operator-framework/operator-controller/blob/13ea200dfa877b268791ccd77089e505ee40fc8d/controllers/operator_controller.go#L54

Reply:
The final name gets prefixed with project name via kustomize, iirc.

Reply:
And if we don't like `manager-role` suffix, we can change it here: https://github.com/operator-framework/operator-controller/blob/13ea200dfa877b268791ccd77089e505ee40fc8d/Makefile#L57

But I'd suggest sticking with the defaults unless we have a good reason not to because that's what other contributors and cluster admins familiar with kubebuilder-based operators will be used to.

Reply:
As of right now, it does not appear that operator-controller is setup with the proper kustomization.
Examples I seeing (e.g. https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/reference/controller-gen.md) are using different name for `rbac:roleName`
Something to do in another PR, but not here.